state-chain/pallets/cf-asset-balances/src/lib.rs

```diff
@@ -136,15 +137,19 @@ impl<T: Config> Pallet<T> {
 		address: ForeignChainAddress,
 		amount: AssetAmount,
 	) -> Result<(), DispatchError> {
-		match T::EgressHandler::schedule_egress(chain.gas_asset(), amount, address.clone(), None)
-			.map_err(Into::into)
-			.and_then(|result @ ScheduledEgressDetails { egress_amount, fee_withheld, .. }| {
-				if egress_amount < REFUND_FEE_MULTIPLE * fee_withheld {
-					Err(Error::<T>::RefundAmountTooLow.into())
-				} else {
-					Ok(result)
-				}
-			}) {
+		match with_storage_layer(|| {
```

Comment:
we can add a unit test for this case too

Reply:
to be clear, I mean the case that if we can't process, that it reverts, there's no egress scheduled, and we still have the amounts etc. in storage.

Reply:
That would be nice, but I think it doesn't make a lot of sense. The only storage that is manipulated here is in the ingress/egress pallet, but in the case of a unit-test it's just the mock that is not using real storage, so I don't think there is a lot to test here. There is already a unit test called `do_not_refund_if_amount_is_too_low` where we check the event and the storage inside the pallet, but that has actually nothing to do with the `with_storage_layer` functionality. I added a check that verifies that the egress mock remains 0.

Reply:
Having a deeper look it is testing the case, but accidentally - and is not testing another case - the case where the schedule_egress itself fails - so we should add one for that case. 

Just to be clear, there are two cases for failure here:
1. The egress amount is less than or equal to the fee (fails inside schedule egress) - though because the mock isn't perfect we would instead just do a `MockEgressHandler::return_failure(true)` to emulate this.
2. The egress *could* be scheduled but we deem it not worthwhile since it's not 100x the amount - this is the case currently hit accidentally. It's accidentally hit because the test sets the fee to double the refund fee - which would normally fail inside the schedule_egress in the usual case.

Reply:
I see it's tested in another place `keep_fees_in_storage_if_egress_fails` so we should be good here actually 👍